### PR TITLE
Nick/s2s implementation

### PR DESCRIPTION
**Pull request by nick-bithead**
Added RTT implementation with functions to enable/disable RTT, register an RTT callback, and send RTT messages although the current test project does not have a test for sending an RTT message at this time. (A chat message can be posted with a normal S2S request)

Added documentation and updated the readme to reflect the new RTT features

### CODE REVIEW

File: S2STest/Source/S2STest/MenuWidget.cpp [hunk 1]
```diff
@@ -12,24 +13,14 @@ void UMenuWidget::NativeConstruct()
 	Super::NativeConstruct();
 
 	//INIT BrainCloud
-	FString appId = "20001";
-	FString serverName = "TestServer";
-	FString serverSecret = "2ddf8355-c516-48dd-a6b0-e35bd75fac80";
+	appId = "26202";
+	FString serverName = "dedicatedserver";
+	FString serverSecret = "d31306d5-ec5d-4be9-b797-84a3a33c8012";
```
Comment (bitheadCody) [inline S2STest/Source/S2STest/MenuWidget.cpp]: Do we want this here?

**Review by bitheadCody — APPROVED** [review]

File: BrainCloudS2S/BrainCloudS2S.cpp [hunk 2]
```diff
@@ -218,16 +229,17 @@ void UBrainCloudS2S::onAuthenticateCallback(const FString &jsonString)
...
         if (_requestQueue.Num())
         {
             _activeRequest = _requestQueue[0];
         }
     }
-    else 
+    else
     {
-        //UE_LOG(LogBrainCloudS2S, Log, TEXT("Number of Waiting Requests, %d"), _requestQueue.Num()); //Check num request in queue
+        UE_LOG(LogBrainCloudS2S, Log, TEXT("Number of Waiting Requests, %d"), _requestQueue.Num()); //Check num request in queue
```
Comment (francobithead) [inline BrainCloudS2S/BrainCloudS2S.cpp]: Needs a if(_logEnabled) condition for logs

File: BrainCloudS2S/BrainCloudS2S.cpp [hunk 3]
```diff
@@ -180,26 +190,27 @@ void UBrainCloudS2S::CheckAuthCredentials(TSharedPtr<FJsonObject> authResponse)
...
     {
         UE_LOG(LogBrainCloudS2S, Error, TEXT("S2S Failed To Authenticate"));
     }
 }
 
-void UBrainCloudS2S::onAuthenticateCallback(const FString &jsonString)
+void UBrainCloudS2S::onAuthenticateCallback(const FString& jsonString)
 {
-    if (_state != State::Authenticated)
+    UE_LOG(LogBrainCloudS2S, Log, TEXT("S2S Authenticate result: %s"), *jsonString);
```
Comment (francobithead) [inline BrainCloudS2S/BrainCloudS2S.cpp]: if(_logEnabled)

File: BrainCloudS2S/BrainCloudS2S.cpp [hunk 4]
```diff
@@ -180,26 +190,27 @@ void UBrainCloudS2S::CheckAuthCredentials(TSharedPtr<FJsonObject> authResponse)
...
         {
-            _sessionId = pData->GetStringField("sessionId");
+            _sessionData.sessionId = pData->GetStringField("sessionId");
         }
-        _heartbeatStartTime = FPlatformTime::Seconds();
-        _state = State::Authenticated;
-        UE_LOG(LogBrainCloudS2S, Log, TEXT("S2S Authenticated"));
+        _sessionData.heartbeatStartTime = FPlatformTime::Seconds();
+        _sessionData.state = S2SState::Authenticated;
+        UE_LOG(LogBrainCloudS2S, Log, TEXT("S2S Authenticated - set heartbeatInterval to: %d"), _sessionData.heartbeatInterval);
```
Comment (francobithead) [inline BrainCloudS2S/BrainCloudS2S.cpp]: if(_logEnabled)

File: BrainCloudS2S/BrainCloudS2S.cpp [hunk 5]
```diff
@@ -180,26 +190,27 @@ void UBrainCloudS2S::CheckAuthCredentials(TSharedPtr<FJsonObject> authResponse)
...
     }
     else
     {
         UE_LOG(LogBrainCloudS2S, Error, TEXT("S2S Failed To Authenticate"));
```
Comment (francobithead) [inline BrainCloudS2S/BrainCloudS2S.cpp]: if(_logEnabled)

**Review by francobithead — APPROVED** [review]
Other than the log comments, it LGTM

**Pull request merged**